lib/s3.py

```diff
@@ -97,18 +98,24 @@ def sign_url(url, method='GET', headers=[], duration=12 * 60 * 60):
 
 
 def main():
+    # to be used like `python3 -m lib.s3 get https://...` from the toplevel dir
     prognam, cmd, uri = sys.argv
 
     url = urllib.parse.urlparse(uri)
     if cmd == 'get':
-        print(sign_url(url))
+        args = sign_curl(url)
+    elif cmd == 'rm':
+        args = ['-XDELETE'] + sign_curl(url, method='DELETE')
     elif cmd == 'put':
-        print(sign_url(url, method='PUT'))
+        args = [sign_url(url, method='PUT')]
```

Comment:
How important is this CLI? Keeping a whole second and outdated signature method in the lib just for this seems not very desirable?

Reply:
I've been finding it pretty useful to be honest, for the purposes of testing.  The main thing that it has over other s3 tools is that it works with our key files in the same way as the other tools.

I'd prefer to keep it.  I will probably eventually port the "put" stuff to include checksums, but right now it's not on the critical path for me...